pkg/globalconfig/global_config.go

```diff
@@ -74,7 +76,13 @@ type GlobalConfig struct {
 	RouterMailpitHTTPSPort           string                      `yaml:"mailpit_https_port,omitempty"`
 	Messages                         MessagesConfig              `yaml:"messages,omitempty"`
 	RemoteConfig                     RemoteConfig                `yaml:"remote_config,omitempty"`
-	// ProjectList is retained to make upgrading easier and should be removed after a few releases
+}
+
+// Old versions of DDEV stored the project list in the global config.
+// For ease of upgrade, we need to be able to extract that data and move it to
+// the new project_list.yaml
+// In a future release, remove this.
+type LegacyGlobalConfig struct {
 	ProjectList map[string]*ProjectInfo `yaml:"project_info"`
 }
```

Comment:
This lets us remove `ProjectList` from the `GlobalConfig` type. It means any time the global config is saved, the project list (if there was one in the file) is removed without having to do something hackish like an above comment points out.

It also gives us a clean way to marshal the list from the YAML file - if we removed `ProjectList` from `GlobalConfig` but *didn't* add something like this, we'd have to parse the file manually which would be even worse.